readme.md

```diff
@@ -1,5 +1,38 @@
 ## Java reactive to Kotlin reactive
 
-- unzip resource file `data.sql.zip`
-- use the VM flag `-XX:+AllowRedefinitionToAddDeleteMethods` if you want to use BlockHound
-- set property to false the second time the application runs: `com.io.reload-db-data=false`
\ No newline at end of file
+1. unzip resource file `data.sql.zip`
+2. use the VM flag `-XX:+AllowRedefinitionToAddDeleteMethods`
+3. set property to false the second time the application runs: `com.io.reload-db-data=false`
+
+## HTTP script
+The project contains the file `scripts/Weather-info-requests.http` that you can use to make api calls directly from your IDE
+
+## Migration to Kotlin
+As a first step, we would like to migrate the project to kotlin. The project is already configured for kotlin but the files are all java.
+
+### Kotlin syntax help
+As some of you are not familiar with kotlin, here are a few tips to help you with the syntax:
+
+- The primitive types and their boxed versions do not exist: `int` and `Integer` become -> `Int` in Kotlin
+- The types all exist in 2 variants: `nullable` and `non-nullable` so the java `String` becomes -> `String` and `String?`
+- every `nullable` variable has to be accessed safely using `?.` instead of `.`: for example: `a.b?.c?.d` -> a is not nullable, but everything after is.
```

Comment:
```suggestion
- every `nullable` variable has to be accessed safely using `?.` instead of `.`: for example: `a.b?.c?.d` -> a is not nullable, but everything after is. It is also possible to perform null checks manually and the value will automatically be cast to non-null version. [See docs](https://kotlinlang.org/docs/null-safety.html)
```